src/main/java/com/openshift/jenkins/plugins/pipeline/Deployment.java

```diff
@@ -148,6 +149,11 @@ public static boolean didAllImagesChangeIfNeeded(String buildConfig, TaskListene
 			
 			if (didImageChangeFromPreviousVersion(client, dc.getLatestVersionNumber(), 
 					chatty, listener, dc.getName(), namespace, latestImageHexID, imageTag)) {
+
+				if(annotateRC) {
+					IReplicationController rc = getLatestReplicationController(dc, namespace, client, chatty ? listener : null);
+    				AnnotationUtils.AnnotateResource(client, listener, chatty, env, rc); 
```

Comment:
so we're only going to end up in this method now(after your change) if someone explicitly marked the build step to "verify deployment is triggered after build completes" which means they at least think they have an imagechangetrigger on the DC.

so i guess that means this is ok as is, we'll annotate deployments that were obviously triggered by the build (deployments for which the image changed), but only if they asked us to check for/verify the deployment explicitly.

and then the other block I referenced below will put the annotation on deployments that were directly triggered by a "deployment" job step.

so all in all I think we're ok once your change goes in.